source/_components/image_processing.tensorflow.markdown

```diff
@@ -0,0 +1,180 @@
+---
+layout: page
+title: "TensorFlow"
+description: "Detect and recognize objects with TensorFlow."
+date: 2018-10-24 00:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: tensorflow.png
+ha_category: Image Processing
+featured: false
+ha_release: 0.82
+---
+
+The `tensorflow` image processing platform allows you to detect and recognize objects in a camera image using [TensorFlow](https://www.tensorflow.org/). The state of the entity is the number of objects detected, and recognized objects are listed in the `matches` attribute. The attribute provides the confidence `score` for recognition and the bounding `box` of the object for each detection category.
+
+## {% linkable_title Setup %}
+TensorFlow is pre-configured in the [official home assistant docker image](https://hub.docker.com/r/homeassistant/home-assistant/).  If you are using this image, you can skip this section.
+
+TensorFlow is special in that it can run on the CPU, or an Nvidia GPU using CUDA.  Because of this, tensorflow is not listed in `REQUIREMENTS`.  In your instance, you will need to pre-install either `tensorflow` or `tensorflow-gpu` via pip:
```

Comment:
```suggestion
TensorFlow is special in that it can run on the CPU, or a Nvidia GPU using CUDA.  Because of this, TensorFlow is not listed in `REQUIREMENTS`.  In your instance, you will need to pre-install either `tensorflow` or `tensorflow-gpu` via pip:
```